docs/src/appendix/snforge-library/fs/file_parser.md

```diff
@@ -0,0 +1,47 @@
+# `FileParser<T>`
+
+```rust
+trait FileParser<T, +Serde<T>> {
+    fn parse_txt(file: @File) -> Option<T>;
+    fn parse_json(file: @File) -> Option<T>;
+}
+```
+
+> ℹ️ **Info**
+>
+> Specific rules must be followed for snforge to correctly parse JSON and plain text files.
+>
+> Read more about them [here](../fs.md#file-format).
```

Comment:
Instead of linking to a general section, let's structure the rules so they are broken down by plain file and json (we pretty much already have that now).

Then we can write these infos to be more specific, e.g. "JSON files must follow specific rules for them to be used in `snforge`, see [link to json section]"